Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1726,9 +1722,11 @@ PhysicalParticleContainer::AddPlasmaFlux (amrex::Real dt)
                     p.id() = -1;
                     continue;
                 }
-
                 Real dens = inj_rho->getDensity(ppos.x, ppos.y, ppos.z);
 
+                const amrex::Real t_fract = amrex::Random(engine)*dt;
```

Comment:
Thanks for this fix. This is the correct thing to do. It seems obvious now, I don't know why I didn't notice this when first implementing this code.

Note that the position update could be moved further down a few lines to after the lines setting up the density. This will skip the update for particles removed because the density is too low.

Also, I notice that `theta` and `xb` will need to be updated to the advanced position since they are used below. Note that for RZ, `xb` will be the radius. It might be easier to completely separate the RZ and XY code below (in their own `ifdef` blocks).

Reply:
Thanks for the review.
I updated the PR and moved `UpdatePosition` further down.
I also removed the variables `xb` and `yb` (which, in practice, are only used for the RZ case) and replaced it by the more explicit variable `radial_position` (which correspond to the `r` position **before** updating the particle position.